src/dotnet/Mgmt.CI.BuildTools/CI/CI.BuildTasks/Mgmt.CI.BuildTasks/BuildTasks/PreBuild/DetectRPScopeTask.cs

```diff
@@ -204,34 +204,39 @@ List<string> GetRPScopes()
                     repoName = tokens[tokens.Length - 1];
                 }
 
-                prFileList = GHSvc.PR.GetPullRequestFileList(repoName, GH_PRNumber);
+                if (GHSvc.IsRepoAuthorized(GitHubRepositoryHtmlUrl))
```

Comment:
While this handles the not-authorized issue did you also want to handle general failures?